### ISSUE TITLE
Inconsistent documentation: self-closing-comp.md

### ISSUE THREAD

**Issue by joshriesenbach-wk** [jsx-eslint/eslint-plugin-react#3852]
Currently in self-closing-comp.md, the first **correct** includes `var contentContainer = <div className="content"></div>;`. Given the rest of that section, the default options, and further examples below this should be in the **incorrect** section or the description for these examples should indicate the `html` option is set to false.

**Comment by vincent067** [comment]
Hi there! 👋 

I took a look at this and you're absolutely right - there's definitely an inconsistency in the documentation.

Looking at the [rule implementation](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/lib/rules/self-closing-comp.js), the default options are:

```js
const optionDefaults = { component: true, html: true };
```

With `html: true` as the default, an empty HTML element like `<div className="content"></div>` **should** be flagged as incorrect (which is why it appears in the **incorrect** section under the `html` option examples). 

But in the first "correct" examples section, this exact code is listed:

```jsx
var contentContainer = <div className="content"></div>;
```

That's confusing for users reading the docs because it contradicts both the default behavior and the later examples. 

I'd be happy to open a PR to fix this! The fix would be to remove that `<div>` example from the first "correct" section since it only belongs in the "incorrect" section when `html: true` (the default). 

Does that sound good? Let me know if you'd prefer a different approach! 🙏

**Comment by ljharb** [comment]
The "help wanted" label means, just make a PR any time.

However, please be very careful with any AI usage - to be copyrightable (and thus viable for an open source contribution) it has to have been done with very heavy human involvement.
